LuaRules/Gadgets/weapon_cluster_ammunition_spawner.lua

```diff
@@ -216,6 +217,49 @@ local function GetRandomAttackPoint(x, z, radius)
 	return target
 end
 
+local function GetRingAttackPoint(x, z, radius, heading)
+	local target = {}
+	target[1] = x + (radius * sin(heading))
+	target[3] = z + (radius * cos(heading))
+	if target[1] > mapx then -- clamp inside the map
+		target[1] = mapx
+	elseif target[1] < 0 then
+		target[1] = 0
+	end
+	if target[3] > mapz then -- ditto
+		target[3] = mapz
+	elseif target[3] < 0 then
+		target[3] = 0
+	end
+	target[2] = spGetGroundHeight(target[1], target[3])
+	return target
+end
+
+local function ConvertProjectileTargetToPos(targettype, targetID, projX, projY, projZ)
+	if targettype == ground then -- this is an undocumented case. Aircraft bombs when targeting ground returns 103 or byte(49).
+		x1 = targetID[1]
+		y1 = targetID[2]
+		z1 = targetID[3]
+		if debug then
+			spEcho(x1,y1,z1)
+		end
+	elseif targettype == 103 then
```

Comment:
This is old code moved up by the looks? If you know what the magic number is maybe add a comment describing it or otherwise?